src/types.ts

```diff
@@ -0,0 +1,137 @@
+/**
+ * These represent the types in our abstract type system.
+ * We will normalize the raw XTP schema into these recursively defined types.
+ */
+
+export type XtpNormalizedKind =
```

Comment:
I'll likely change the name of this. This is just so we don't conflict with XtpType.